Sources/Widgets/Core/DragHelper/index.js

```diff
@@ -0,0 +1,44 @@
+import macros from 'vtk.js/Sources/macros';
+import { add, subtract } from 'vtk.js/Sources/Common/Core/Math';
+
+function vtkWidgetDragHelper(publicAPI, model) {
+  model.classHierarchy.push('vtkWidgetDragHelper');
+
+  publicAPI.calcDelta = (dragWorldCoords) => {
+    if (dragWorldCoords?.length !== 3 || !model.lastOrigin) return [0, 0, 0];
+    return subtract(dragWorldCoords, model.lastOrigin, [0, 0, 0]);
+  };
+
+  publicAPI.calcPosition = (position, dragWorldCoords) => {
+    // if no position is given, default to the dragging world coords
+    if (!position) return dragWorldCoords;
+    const delta = publicAPI.calcDelta(dragWorldCoords);
+    return add(position, delta, [0, 0, 0]);
+  };
+
+  const origSetLastOrigin = publicAPI.setLastOrigin;
+
+  publicAPI.setLastOrigin = (origin) => {
+    // allow setting null as the last origin
+    if (origin && origin.length !== 3) return false;
+    return origSetLastOrigin(origin);
+  };
+}
+
+const defaultValues = (initialValues) => ({
+  lastOrigin: null,
```

Comment:
maybe rename into lastHandleOrigin or lastWidgetOrigin ?

Personal preference: I prefer "currentOrigin" vs "lastOrigin". For some reason working with a "last" information carries a notion of "obsolete" information.